docs/contributor-guide/how to/rule.md

```diff
@@ -0,0 +1,226 @@
+# Develop a rule
+
+A `rule` is a check that `sonarwhal` will validate. The API should be
+flexible enough to allow you to implement anything you want easily:
+
+* Validate that all links are `HTTPS`.
+* Integrate with a third party service.
+* Inject JavaScript to execute in the context of the page.
+* etc.
+
+If there is something you want to do and you can’t, or it is not clear
+how to do it, please open an issue.
+
+## Types of rules
+
+There are 2 types of `rule`s a user can develop:
+
+* `external`: These are `rule`s that are published independently. When a
+  `rule` is specific to a domain or use case, it should be external.
+* `core`: These are the `rule`s that are shipped with `sonarwhal` directly.
+  Before starting to develop a `core rule`, please make sure there is
+  an open issue and talk with the maintainers about it.
+
+Both types of `rule`s [work exactly the same](#howruleswork), the only
+difference being where they are located.
+
+### Creating an external rule
+
+The easiest wait to create a new rule that will be distributed outside
+`sonarwhal` is via the CLI parameter `--new-rule`. You have 2 options:
```

Comment:
Maybe use _"several options such as:"_ instead of "2 options"? You can also install `sonarwhal` locally and use it via `npm` scripts.

Reply:
I wasn't sure about updating this section. It will be obsolete when you finish the refactor work that should happen in the next few days so...

Reply:
Sure, just update "several options such as:", and leave the rest as it is, and will change it later.